### PR TITLE
DIP-267 Provide a User Data mechanism for links to profiles

### PR DESCRIPTION
**Pull request by wesbiggs**
Problem
=======
#267 discusses the advantages of DSNP profile documents being discoverable based on a user's identifier, rather than being temporal data that must be externally tracked and indexed to be useful.

Solution
========
This proposal adds a new User Data type, `profileResources`, which is currently defined for the Activity Content profile document type, but is extensible to other document types that may need to be user-centric in the future.
A profile resource is simply a link type (as an integer) and a CID as a string plus byte length.
We define link type 1 for AC profiles.

This change assumes we are versioning for 1.3, but does not include versioning updates, which are in other open PRs at this time.

Change summary:
---------------
- [ ] ~~Updated Spec Versions~~
- [x] Added definition for ProfileResource Avro type
- [x] Added entry in User Data page for `profileResource`
- [x] Updated navigation; added ProfileResource, moved Profile announcement to "Migrated" section

### CODE REVIEW

File: pages/DSNP/Identity.md [hunk 1]
```diff
@@ -29,18 +29,29 @@ An implementation MUST NOT allow the user to remove the only or last remaining c
...
-* An owner MUST be able to delegate permission to announce on their behalf to other parties.
+* A user MUST be able to delegate and revoke permissions to other parties to submit each type of announcements or replace each type of user data on their behalf.
 * A user MUST be able to revoke delegated permissions.
-* Announcements from a delegate MUST be able to be verify which delegate made the specific Announcement.
+* Announcements and user data updates from a delegate MUST be able to be verified as to which delegate was responsible for the action.
 * Delegation revocation MUST NOT be retroactive.
 
+## Identity-Linked State Data
+
+DSNP identities can have linked [User Data](UserData.md), which, like the identifier and delegations,  forms part of the shared state of the consensus system.
```
Comment (wilwade) [inline pages/DSNP/Identity.md:46]: Random extra space.
```suggestion
DSNP identities can have linked [User Data](UserData.md), which, like the identifier and delegations, forms part of the shared state of the consensus system.
```

File: pages/DSNP/Types/ProfileResource.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "doc": "Profile-linked resource",
+    "fields": [
+        {
+            "name": "type",
+            "type": "int",
+            "doc": "Type of resource"
+        },
+        {
+            "name": "cid",
+            "type": "string",
```
Comment (wilwade) [inline pages/DSNP/Types/ProfileResource.md:23]: Is a string the best way to represent a CID in Avro?

Either way we should be more clear how to serialize a CID into a string.

Specifically this should be a `cidv1` I believe

**Review by wilwade — COMMENTED** [review]
One clarification needed about CID serialization, but otherwise looks like a good improvement to me!

File: pages/DSNP/Identifiers.md [hunk 3]
```diff
@@ -49,11 +49,29 @@ Any [Announcement Types](Announcements.md#announcement-types) with a `fromId` an
...
-dsnp://78187493520/0x1234567890abcdef0123456789abcdef0123456789abcdef0123456789abcdef
+dsnp://78187493520/QmQNHNfHnbgJJ6nK4UPx2VtTUCafAKCbqZJ6ZRYUGjoeFj
 ```
 
 | part | value |
 | ---- | ----- |
 | Scheme | `dsnp://` |
 | User Id | `78187493520` |
-| Content Hash | `0x1234567890abcdef0123456789abcdef0123456789abcdef0123456789abcdef` |
+| Content Hash | `QmQNHNfHnbgJJ6nK4UPx2VtTUCafAKCbqZJ6ZRYUGjoeFj` |
```
Comment (wilwade) [inline pages/DSNP/Identifiers.md:59]: I believe this change should also propagate to `pages/ActivityContent/Associated/Hash.md` Or would that require https://github.com/LibertyDSNP/spec/issues/273 to be implemented first?

Comment (wesbiggs) [reply pages/DSNP/Identifiers.md:59]: This PR is just changing it to a multihash multibase string, as we don't serialize hashes as hex anywhere as far as I know. (The DSNP URI should match the hash from the announcement, which is now a multihash.)

Technically unrelated to the purpose of this PR, just a vestigial issue I happened to run into. I can make it a separate PR if you think it's warranted.

Comment (wilwade) [reply pages/DSNP/Identifiers.md:59]: It is fine. Just checking

**Review by wilwade — APPROVED** [review]
Nice improvement. Matches expected DIP-267

**Pull request merged**